src/semafour.cc

```diff
@@ -257,6 +325,50 @@ void Semafour::Close(const v8::FunctionCallbackInfo<v8::Value>& args) {
 }
 
 
+void Semafour::TryWait(const v8::FunctionCallbackInfo<v8::Value>& args) {
+  Semafour* sem = UNWRAP(Semafour);
+  Isolate* isolate = args.GetIsolate();
+  int r;
+
+  if (args.Length() == 0) {
+    // Handle synchronous case.
+    r = sem->TryWait();
+
+    if (r != TRYWAIT_LOCKED && r != TRYWAIT_OK) {
```

Comment:
I think this would just be `if (r != EAGAIN) {`

Reply:
I don't think so: this would run `THROW_UV` if `r` was `0` (`kTryWaitOk`), which is a perfectly valid code meaning that the lock was acquired successfully.